feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/rrm_config.c

```diff
@@ -24,11 +24,122 @@
 #include "uci.h"
 #include "utils.h"
 #include "captive.h"
+#include <fcntl.h>
 
 ovsdb_table_t table_Wifi_RRM_Config;
 extern ovsdb_table_t table_Wifi_Radio_Config;
 extern ovsdb_table_t table_Wifi_VIF_Config;
 
+bool rrm_config_txpower(const char *rname, unsigned int txpower)
+{
+	char cmd[126] = {0};
+	int rid = 0;
+	txpower = txpower * 100;
+
+	sscanf(rname, "radio%d", &rid);
+
+	memset(cmd, 0, sizeof(cmd));
+	/* iw dev <devname> set txpower <auto|fixed|limit> [<tx power in mBm>]*/
+	snprintf(cmd, sizeof(cmd),
+		"iw phy phy%d set txpower fixed %d", rid, txpower);
+	system(cmd);
+
+	return true;
+}
+
+#define NUM_OF_RATES 12
+bool rrm_config_mcast_bcast_rate(const char *ifname, char *band,
+				 unsigned int bcn_rate, unsigned int mcast_rate)
+{
+	int fd, i, l;
+	char phy[32] = {0};
+	char path[126] = {0};
+	char value[126] = {0};
+	unsigned int bcn_code = 0xFF;
+	unsigned int mcast_code = 0xFF;
+	unsigned int band_id = 0;
+	unsigned int rate_code[NUM_OF_RATES][2] = {{1,0x43}, {2,0x42}, {5,0x41},
+						  {11,0x40}, {6,0x3}, {9,0x7},
+						  {12,0x2}, {18,0x6}, {24,0x1},
+						  {36,0x5}, {48,0x0}, {54,0x4}};
+
+	if (!strcmp(band, "2.4G"))
+		band_id = 2;
+	else
+		band_id = 5;
+
+	/* beacon rate given by cloud in multiples of 10 */
+	if (bcn_rate > 0)
+		bcn_rate = bcn_rate/10;
+
+	/* get phy name */
+	snprintf(path, sizeof(path), "/sys/class/net/%s/phy80211/name", ifname);
```

Comment:
Make the phy from ifname a function of it's own for readability.

Reply:
done